components/number/template.rst

```diff
@@ -10,14 +10,32 @@ using :ref:`lambdas <config-lambda>`.
 
 .. code-block:: yaml
 
-    # Example configuration entry
+    # Example configuration entry for setting a value, either in combination with the set_action configuration or the add_on_state_callback() method in a custom component.
```

Comment:
I don't like this addition exposing information for custom components here.
Anyone developing custom components should be looking at the api docs themselves, not the configuration docs.